mlrun/feature_store/retrieval/job.py

```diff
@@ -37,17 +44,20 @@ def run_merge_job(
         raise mlrun.errors.MLRunInvalidArgumentError("target object must be specified")
     name = f"{name}_merger"
     run_config = run_config or RunConfig()
+    kind = run_config.kind or ("spark" if engine == "spark" else "job")
     if not run_config.function:
         function_ref = vector.spec.function.copy()
         if function_ref.is_empty():
-            function_ref = FunctionReference(name=name, kind="job")
+            function_ref = FunctionReference(name=name, kind=kind)
         if not function_ref.url:
-            function_ref.code = _default_merger_handler
+            function_ref.code = _default_merger_handler.replace(
+                "{{{engine}}}", merger.__name__
+            ).replace("{{{engine_args}}}", str(engine_args or {}))
```

Comment:
I think it's better and cleaner to pass the `engine_args` in the `task` - either expanding `**engine_args` into the task dictionary, or passing the dict as a new param `"engine_args": engine_args`.